spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/BootstrapConfiguration.java

```diff
@@ -49,8 +53,10 @@
 
 		@Bean
 		@ConditionalOnProperty(name = "spring.cloud.kubernetes.config.enabled", matchIfMissing = true)
-		public ConfigMapPropertySourceLocator configMapPropertySourceLocator(ConfigMapConfigProperties properties) {
-			return new ConfigMapPropertySourceLocator(this.client, properties);
+		public ConfigMapPropertySourceLocator configMapPropertySourceLocator(ConfigMapConfigProperties properties,
```

Comment:
Instead of the `Optional` here I think it might be cleaner if you used `@ConditionalOnMissingClass`